draft-ietf-teep-protocol.md

```diff
@@ -642,6 +642,7 @@ update = [
     ? manifest-list => [ + bstr .cbor SUIT_Envelope ],
     ? attestation-payload-format => text,
     ? attestation-payload => bstr,
+    ? err-code => uint .size 1,
```

Comment:
The TEEP error message has `err-code: 0..23`.   Is this the same numbering space or a different one?  Why is the type different if the same? 

The TEEP error message also has `err-msg => text .size (1..128),` and if we add an err-code here, shouldn't we also have an err-msg?

Reply:
In cbor, the 'uint .size 1' and '0..23' is the same representation in binary. 
The one byte unsigned integer could only take 0 to 23 in cbor.
I just used 'uint .size 1' here just because there was the same representation above in the same cddl file.
I will check the cbor syntax with the cddl tool to see which is more cddl tool friendly.

Reply:
For the err-msg, I will add it.
Please wait for my updates on this PR.

Reply:
@dthaler Added the err-msg.
Please review.

Reply:
> In cbor, the 'uint .size 1' and '0..23' is the same representation in binary. The one byte unsigned integer could only take 0 to 23 in cbor. I just used 'uint .size 1' here just because there was the same representation above in the same cddl file. I will check the cbor syntax with the cddl tool to see which is more cddl tool friendly.

Please update the CDDL to use the same type for err-code throughout.  Currently the CDDL in the draft has:
```
check-draft-ietf-teep-protocol.cddl:  err-code: 0..23
check-draft-ietf-teep-protocol.cddl:; The err-code parameter, uint (0..23)
```
So either change those, or else use 0..23 instead of uint size 1.

Reply:
Unified them to `uint size 1` in cddl syntax and revised the comments that they consume one byte for numbers 0..23 and do not take up to 23 bytes.